examples/modal/modal.go

```diff
@@ -149,6 +173,22 @@ func handleModal(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusUnauthorized)
 		return
 	}
+
+	// update modal sample
+	switch i.Type {
+	//update when interaction type is view_submission
+	case slack.InteractionTypeViewSubmission:
```

Comment:
I couldn't get `api.UpdateView` to work under this event. I had to use `slack.InteractionTypeBlockActions` e.g `block_action`. Is this working for you?

Reply:
Thank you for the feedback,I would check that.

Reply:
It's working but modal closes right after update.

I think this is caused by slack server.
after update modal,app respond http status 200 to slack server and the server see it as over

I'v been running this code local and connect via ngrok and this happen

I also run another app on AWS Lambda using this logic but it doesn't provide the same result,woring fine

Reply:
by adding `time.Sleep(time.Second * 2)` after api.UpdateView you can prevent sending http 200 and can see updated modal.

```go
	case slack.InteractionTypeViewSubmission:
		//you can use any modal you want to show to users just like creating modal.
		updateModal := updateModal()
		// You must set one of external_id or view_id and you can use hash for avoiding race condition.
		// More details: https://api.slack.com/surfaces/modals/using#updating_apis
		_, err := api.UpdateView(updateModal, "", i.View.Hash, i.View.ID)
		time.Sleep(time.Second * 2)
		if err != nil {
			fmt.Printf("Error updating view: %s", err)
			w.WriteHeader(http.StatusInternalServerError)
			return
		}
```

Reply:
@kanata2 @charlesoconor 
sorry for kept you waiting,could you review this code?